src/hb-mutex.hh

```diff
@@ -81,20 +81,26 @@ typedef int hb_mutex_impl_t;
 
 #else
 
-#error "Could not find any system to define mutex macros."
-#error "Check hb-mutex.hh for possible resolutions."
+#include <mutex>
+#include <new>
+typedef std::mutex              hb_mutex_impl_t;
+#define hb_mutex_impl_init(M)   HB_STMT_START { new (M) hb_mutex_impl_t; } HB_STMT_END
```

Comment:
This is also a bit squirrel-y, technically one needs to access the object from the pointer returned from `new`, the original pointer passed in to `new` is not blessed with "objectness". Technically, a compiler is allowed to optimize this `new` away, since the resulting object can never be accessed. I can't store it in `m`  since that's actually an array, not a pointer.

Reply:
I'm fairly sure that's not the case for placement new. Trying to find some reference to that.

Reply:
Here, for example: https://en.cppreference.com/w/cpp/memory/allocator_traits/construct
We can call that if you prefer.

Reply:
It seems even stranger to use `std::allocator_traits<Alloc>::construct` since the underlying memory was not allocated with any allocator, and `construct` is spec'ed to call placement new anyway. The reality is that I don't think any compiler can really take advantage of not using the return value of placement new since it would break so much existing code (even if it is technically undefined behavior).

So mostly, I think this is probably ok, it just makes me uncomfortable. Essentially every vector like container has this issue.